shapely/_geometry.py

```diff
@@ -104,13 +104,16 @@ def get_dimensions(geometry, **kwargs):
 
     Examples
     --------
-    >>> get_dimensions(Geometry("POINT (0 0)"))
+    >>> from shapely import GeometryCollection, Point, Polygon
+    >>> point = Point(0, 0)
+    >>> get_dimensions(point)
     0
-    >>> get_dimensions(Geometry("POLYGON((0 0, 0 10, 10 10, 10 0, 0 0))"))
+    >>> polygon = Polygon(((0, 0), (0, 10), (10, 10), (10, 0), (0, 0)))
```

Comment:
The linestring above (line 82) uses a _list_ of coordinate pair tuples, while here it is a _tuple_ of tuples. Both work fine, but we should maybe try to be consistent within the docs? 
I personally find the list of tuples more readable (otherwise you have all those same type of brackets):

```suggestion
    >>> polygon = Polygon([(0, 0), (0, 10), (10, 10), (10, 0), (0, 0)])
```

(this also doesn't necessarily need to happen in this PR, as it is quite some work I can imagine)

Reply:
I'd prefer to leave that for a follow-on PR to do just that, in order to keep that both changesets a bit smaller.